src/disk.hpp

```diff
@@ -42,45 +41,42 @@ class Disk {
 
     virtual std::string GetFileName() = 0;
 
-    virtual ~Disk(){};
+    virtual ~Disk() = default;
 };
 
-class FileDisk : public Disk {
-public:
-    inline explicit FileDisk(const fs::path &filename)
+struct FileDisk : Disk {
+    explicit FileDisk(const fs::path &filename)
     {
         filename_ = filename;
 
         // Opens the file for reading and writing
-        f_ = fopen(filename.c_str(), "w+b");
+        f_ = ::fopen(filename.c_str(), "w+b");
+        if (f_ == nullptr) {
+            throw InvalidValueException(
+                "Could not open " + filename.string() + ": " + ::strerror(errno));
```

Comment:
always report a failure to open a file (in the future, perhaps we should do something similar to when reading or writing fails, to wait 5 minutes and retry).